add orientation and better compatible with coord_flip

xiangpin opened the pull request:
+ The ggplot2 has introduced `orientation` to control the orientation of the layer (either "x" or "y"). This is useful because some geometric layers can not change the "x" or "y" in the `mapping`, such as `geom_density_ridges`.
+ In addition, the layer can also use `orientation` to directly control the orientation of the layer, such as `geom_col`, `geom_bar`, `geom_bar` ... But, when it is also compatible with the `coord_flip`.

## original

```
library(ggplot2)
library(ggsignif)
dat <- data.frame(
            Group = c("S1", "S1", "S2", "S2"),
            Sub = c("A", "B", "A", "B"),
             Value = c(3, 5, 7, 8)
            )
p1 <- ggplot(dat, mapping=aes(x=Group, y=Value)) +
             geom_col(aes(fill=Sub), position="dodge", width=0.5) + 
             geom_signif(y_position=c(5.3, 8.3), 
                                 xmin=c(0.8, 1.8), 
                                 xmax=c(1.2, 2.2),
                                 annotation=c("**", "NS"), 
                                 tip_length=0) +
             geom_signif(comparisons = list(c("S1", "S2")),
                                  y_position = 9.3, 
                                  tip_length = 0, 
                                  vjust = 0.2) + 
             scale_fill_manual(values = c("grey80", "grey20"))
p1
``` 
![p1](https://user-images.githubusercontent.com/17870644/128282691-fb644393-ca22-492b-aa20-b49087390a2b.PNG)

## Introducing orientation
```
p2 <- ggplot(dat, mapping=aes(y=Group, x=Value)) +
          geom_col(aes(fill=Sub), position="dodge", width=0.5, orientation="y") + 
          geom_signif(y_position=c(5.3, 8.3), 
                              xmin=c(0.8, 1.8), 
                              xmax=c(1.2, 2.2),
                              annotation=c("**", "NS"), 
                              tip_length=0, 
                              orientation="y") +
          geom_signif(comparisons = list(c("S1", "S2")),
                               y_position = 9.3, 
                               tip_length = 0, 
                               vjust = 0.2, 
                                orientation="y") + 
          scale_fill_manual(values = c("grey80", "grey20"))
p2
```
![p2](https://user-images.githubusercontent.com/17870644/128282849-82e0dd37-d0af-4562-98e2-1188ce82b063.PNG)

## Orignal + coord_flip

```
p3 <- p1 + coord_flip()
p3
```
![p3](https://user-images.githubusercontent.com/17870644/128283032-b652ae08-5701-4285-857e-686fda3e0f8b.PNG)

## orientation + coord_flip

```
p4 <- p2 + coord_flip()
p4
```
![p4](https://user-images.githubusercontent.com/17870644/128283041-70d8d555-c07b-4fa3-9601-2f3290da9347.PNG)

IndrajeetPatil commented:
Thank you so much for working on this! This looks fantastic.

Just a couple of minor things:
- Why are you adding a Makefile? Is it necessary?
- Can you please make an entry in `NEWS.md` about this change? 

Thanks

const-ae commented:
@xiangpin, Wow this is really great, thank you for addressing this longstanding shortcoming.

@IndrajeetPatil, do you know what is going on with all the CI failures. Is that expected?

> Can you please make an entry in NEWS.md about this change?

That's a good idea. In addition, I would suggest adding an example in the README.Rmd and also if possible some unit tests

IndrajeetPatil commented:
Some of the tests need to be updated because of this change. I can updated once this is merged.

xiangpin commented:
Hello @IndrajeetPatil @const-ae 

I have made some updates according to your suggestions.

+ The `Makefile` is useful for developers to `build` or `check` ... on the local computer since it can save time by collecting common commands. For example, `make all` can replace the `R CMD build ...`, `R CMD check ...` etc. in this file.
+ I have updated the `NEWS.md`, added an example to the `README`, and modified the unit test.

best wishes
xushuangbin

IndrajeetPatil commented:
Excellent. Thanks once again for your contribution! :)
